ludwig/datasets/mixins/download.py

```diff
@@ -57,17 +65,27 @@ def download_raw_dataset(self):
         Download the raw dataset and extract the contents of the zip file and
         store that in the cache location.
         """
-        os.makedirs(self.raw_temp_path, exist_ok=True)
-        for url in self.download_urls:
-            with urlopen(url) as zipresp:
-                with ZipFile(BytesIO(zipresp.read())) as zfile:
-                    zfile.extractall(self.raw_temp_path)
-        os.rename(self.raw_temp_path, self.raw_dataset_path)
+        #os.makedirs(self.raw_temp_path, exist_ok=True)
+
+        with tempfile.TemporaryDirectory() as tmpdir:
```

Comment:
Can we simplify to something like this?

```
with fs_utils.upload_output_file(self.raw_dataset_path) as tmpdir:
    ...
```

If that doesn't work, we should still make something like this in `fs_utils` as we do the same thing below.